### DISCUSSION TITLE
Enhancement: custom song tgat will be editer and design draw.

### DISCUSSION THREAD

**Discussion by GOOBOZA** [FunkinCrew/Funkin#1681] [Ideas]
what if there is an ingame editer that makes it so you cna have an osu and custom song. there is also an custom drawing editer that makes it so you can do a lot of mod charts. and there will be a coding   section that will make it a coding.

**Comment by GOOBOZA** [comment]
and there is an also will be that might be some ingame song and there are gonna be week 8

**Comment by GOOBOZA** [comment]
also there will be the weeks that are release

**Comment by inue-kang** [comment]
Ok, first off, there is already a song charting tool built into FNF, osu is irrelevant, and what? You can already modify code. Just learn how to mod.

**Comment by owoellen** [comment]
osu isn’t irrelevant

**Comment by GOOBOZA** [comment]
bruh im saying an ingame editer tool that are will be mod chart not just chart and you will be upload it to the server. osu is not iirelavent because i am saying the charting will be support osu converter and you are charting in osu and then convert to fnf.
what i mean by the code modifying is i am modifying than i do lime test windows. and then if i download it. it will be 100mb and i dont have enough storage for that and its a problem. and they will make it not hard coding. so you can do it in game.

**Comment by GOOBOZA** [comment]
> Ok, first off, there is already a song charting tool built into FNF, osu is irrelevant, and what? You can already modify code. Just learn how to mod.

**Comment by inue-kang** [comment]
If you are looking for a mod charting tool, just use MIDItoFNF tool by NyxtheShield.  Otherwise, just use the ingame charting tool but change the music to your mod music and hand chart.  As for osu converter and fnf charting, how would you even do that? You have to convert circle positions to arrow types. I don't think thats possible.

**Comment by inue-kang** [comment]
And don't worry too much, ninjamuffin99 stated that they will make modding easy and built in-game, its just gonna be a while.

**Comment by GOOBOZA** [comment]
oh ok thank you for the mod charting tool but i mean osu mania were its like an chart like this: https://www.youtube.com/watch?v=602IcfqT5Fw&ab_channel=staiain i would want to convert to a json

**Comment by inue-kang** [comment]
oh, osu mania, hmmm, idk how that game works

**Comment by libauror-a** [comment]
ig use this to convert osumania to fnf json?
https://github.com/ItzFireable/mania2fnf

**Comment by steviegt6** [comment]
> oh, osu mania, hmmm, idk how that game works

It's literally just a 4key Mania game.

**Comment by inue-kang** [comment]
> 4key

I know that, what i meant was i don't know what coding language the game uses, what the code is, how it displays stuff like the arrow movement, how it knows which arrows to display, etc.

**Reply by steviegt6** [reply]
osu! uses C# and Microsoft's XNA framework, though it likely uses a modernized "version" like MonoGame or FNA now.
osu!lazer uses the osu!framework, but that's still a work-in-progress. It's also in C#.

**Comment by isakube** [comment]
stroke
